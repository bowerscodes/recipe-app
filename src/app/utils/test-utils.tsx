import { ReactNode } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { HeroUIProvider } from "@heroui/react";
import RecipeProvider from "@/context/RecipeContext";

const AllProviders = ({ children }: { children: ReactNode }) => (
  <HeroUIProvider>
    <RecipeProvider>{children}</RecipeProvider>
  </HeroUIProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
