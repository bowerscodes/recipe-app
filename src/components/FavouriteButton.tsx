import { FaHeart, FaRegHeart } from "react-icons/fa";

import { useRecipeContext } from "@/context/RecipeContext";
import { Recipe } from "@/types/Recipe";

export default function FavouriteButton({ recipe }: { recipe: Recipe }) {
  const { toggleFavourite, isFavourite } = useRecipeContext();
  const favourited = isFavourite(recipe.id);

  return (
    <span
      onClick={(e) => {
        e.stopPropagation();
        toggleFavourite(recipe.id);
      }}
      onKeyDown={e => {
        if (e.key === "Enter" || e.key === " ") {
          e.stopPropagation();
          toggleFavourite(recipe.id);
        }
      }}
      className="p-1 cursor-pointer"
      role="button"
      tabIndex={0}
      color={`${favourited ? "#ef4444" : "black"}`}
      aria-label={
        favourited
          ? `Remove ${recipe.title} from favourites`
          : `Add ${recipe.title} to favourites`
      }
    >
      {favourited ? (
        <FaHeart style={{ color: "#ef4444" }} />
      ) : (
        <FaRegHeart style={{ color: "black" }} />
      )}
    </span>
  );
}
