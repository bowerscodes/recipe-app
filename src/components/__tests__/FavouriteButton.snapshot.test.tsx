import { render } from "@/app/utils/test-utils";
import FavouriteButton from "../FavouriteButton";
import recipesData from "@/data/recipes.json";

const recipe = recipesData[0];

describe("FavouriteButton Snapshots", () => {
  test("renders FavouriteButton in unfavourited state", () => {
    const { container } = render(<FavouriteButton recipe={recipe} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders FavouriteButton with user recipe", () => {
    const userRecipe = {
      id: 1001,
      title: "My Recipe",
      image: "",
      cookingTime: "20 minutes",
      ingredients: ["ingredient 1"],
      instructions: ["step 1"],
      reviews: [],
      author: "me"
    };

    const { container } = render(<FavouriteButton recipe={userRecipe} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
