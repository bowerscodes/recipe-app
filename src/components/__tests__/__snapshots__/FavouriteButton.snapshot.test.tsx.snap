// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FavouriteButton Snapshots renders FavouriteButton in unfavourited state 1`] = `
<div>
  <span
    aria-label="Add Lasagne & Chips to favourites"
    class="p-1 cursor-pointer"
    color="black"
    role="button"
    tabindex="0"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: black;"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
      />
    </svg>
  </span>
</div>
`;

exports[`FavouriteButton Snapshots renders FavouriteButton with user recipe 1`] = `
<div>
  <span
    aria-label="Add My Recipe to favourites"
    class="p-1 cursor-pointer"
    color="black"
    role="button"
    tabindex="0"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: black;"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
      />
    </svg>
  </span>
</div>
`;
